### ISSUE TITLE
maxing out ram (fixed)

### ISSUE THREAD

**Issue by ghost** [ModdedMinecraft/MMCRestrict#4]
i had this really weird issue with this plugin by maxing out the ram after using this pluggin

sadly there is no log to trace but after removal of MMcRistrct all went back to normal any idea where its comming from because the plugin on its own is great

so im hoping you know this issue and how to fix it
 
its going about a FTB Beyond 1.10.2 pack version 1.10.0 server running on spongeforge 2477

**Comment by leelawd** [comment]
Unfortunately I have not came across this issue. But i do have a suspicion on why it may be happening.

Could you send me the mmcrestrict.conf and banneditems.conf files, I may need them to run some tests.

Also if you know the full java version and OS the server is running on, that may be helpful.

**Comment by baconetworks** [comment]
java 8 os is unkown hosted server probbally centOs

ill check if i still have those files and will up load them

**Comment by baconetworks** [comment]
https://pastebin.com/whk4qekL banneditems
https://pastebin.com/kr5W6QrH mmcr

**Comment by SubtypeZero** [comment]
This may be a completely separate issue, but during times of severe lag I have ran some profiles and this seems to really stick out. MMCRestrict is way up there in the sample I took, much higher than other mods (which I would have expected to be taking the most amount of processing time). Take a look at this [screenshot](https://images-ext-2.discordapp.net/external/gbH5WFfbQhzLpSnE90B-QYUh9-kbYm2VlTeD4VfbryE/https/i.gyazo.com/dc8db01f94609f7d510b533dfde06360.png?width=969&height=677).

**Comment by Carhenia** [comment]
I've had the same issue, however with my CPU. I kept getting the same error on the console, saying that the server was lagging behind and couldn't keep up. This only started after I began working on my blacklist, so when I tested it by removing MMCrestrict, my cpu count went down to 20% and the console was no longer giving me the error. I really hope this is a fixable issue, but for the moment I can't use it since it makes my server unplayable.

**Comment by baconetworks** [comment]
This exactly my problem

**Comment by Carhenia** [comment]
Glad its not just me, was scared it was one of my mods and I would have to figure out which... Hope to hear back from you @leelawd, I can send those files aswell if needed.

**Comment by leelawd** [comment]
@Carhenia was there any pattern in the rate that these could not keep up errors were appearing?

**Comment by leelawd** [comment]
I believe i have found the cause of the issue. Unfortunately I am unable to fully test whether this ultimately fixes the problem and would greatly appreciate if you could test this build for me.

All of the changes are here https://github.com/ModdedMinecraft/MMCRestrict/commit/16852dbd5a115c43c0fb45fda827147656eaef0e

Basically i have removed the automatic chunk searching and replaced it with a manual command instead of running constantly.

Build 1.3.2:
https://drive.google.com/file/d/0B5kVUsw24he0OUtXbm5wTTZIem8/view?usp=sharing

**Comment by baconetworks** [comment]
@leelawd im implemetning your fix now sorry haddnt much time ill let you know how it goes

**Comment by Carhenia** [comment]
Forgot to respond to this, I apologise. The fix worked, CPU levels are back to normal, sometimes still get the 'can't keep up' message on the console, but only once or twice, thanks for your time :)

**Comment by baconetworks** [comment]
the cpu usage and ram usage was the main issue

**Comment by baconetworks** [comment]
@leelawd atm the plugin seems stable no wierd spikes like the last time

**Comment by baconetworks** [comment]
after an hour of uptime the server starts lagging again ill post hte files in a min

**Comment by baconetworks** [comment]
issue is been resolved after doing my black list just a few spikes but stopped after that thanks for the fix

**Comment by unstablecraft** [comment]
@leelawd  this sane issue appears on api7

**Comment by leelawd** [comment]
The updated version has been put on Ore as 1.4.0. 

https://ore.spongepowered.org/leelawd93/MMCRestrict/versions
